### PR TITLE
Malleus improvements

### PR DESCRIPTION
**Pull request by JonathanLennox**
This is a cleaned-up version of @gpolitis 's malleus patches, along with some enhancements by @bgrozev .

### CODE REVIEW

File: src/test/java/org/jitsi/meet/test/MalleusJitsificus.java [hunk 1]
```diff
@@ -101,29 +119,31 @@
...
         {
             runThreads[i]
                 = runAsync(
-                    i,
                     url,
                     waitTime,
-                    i >= numSenders /* no video */);
+                    i >= numSenders /* no video */,
+                    i >= numAudioSenders /* no audio */,
+                    regions == null ? null : regions[i % regions.length]);
```
Comment (bbaldino) [inline src/test/java/org/jitsi/meet/test/MalleusJitsificus.java]: @bgrozev do we want to add support for the "stacked sender" allocation as well as the evenly distributed one here?

Comment (JonathanLennox) [reply src/test/java/org/jitsi/meet/test/MalleusJitsificus.java]: If we do we should probably split it out into a separate PR - in fact I'm wondering if @bgrozev's existing enhancements from this PR should be split out as well.

Comment (bgrozev) [reply src/test/java/org/jitsi/meet/test/MalleusJitsificus.java]: I just modified the code in place for the "stacked sender" test, I don't have anything ready for committing.

**Comment by gpolitis** [comment]
Could you please cherry-pick this https://github.com/jitsi/jitsi-meet-torture/pull/257/commits/1855c84c0d43ba2eb9522717bfe3ef5e0ef634b3 ? It removes some obsolete additions/modifications (for example the joinNextParticipant method that isn't used anywhere).

**Comment by JonathanLennox** [comment]
I already cherry-picked ` 1855c84` - it's squashed into the "Improve the reliability of Malleus" commit.

` 71ec571` doesn't apply cleanly, possibly because of that squash.  Can you apply it on this branch?

**Comment by gpolitis** [comment]
@JonathanLennox this should apply cleanly
 
```
diff --git a/src/test/java/org/jitsi/meet/test/MalleusJitsificus.java b/src/test/java/org/jitsi/meet/test/MalleusJitsificus.java
index a737590..1ee55ad 100644
--- a/src/test/java/org/jitsi/meet/test/MalleusJitsificus.java
+++ b/src/test/java/org/jitsi/meet/test/MalleusJitsificus.java
@@ -139,6 +139,7 @@ public class MalleusJitsificus
         {
             runThreads[i]
                 = runAsync(
+                    i,
                     url,
                     waitTime,
                     i >= numSenders /* no video */,
@@ -155,7 +156,8 @@ public class MalleusJitsificus
         }
     }
 
-    private Thread runAsync(JitsiMeetUrl url,
+    private Thread runAsync(int i,
+                            JitsiMeetUrl url,
                             long waitTime,
                             boolean muteVideo,
                             boolean muteAudio,
@@ -183,7 +185,7 @@ public class MalleusJitsificus
                 _url.appendConfig("config.deploymentInfo.userRegion=\"" + region + "\"");
             }
 
-            WebParticipant participant = joinNextParticipant(_url, ops);
+            WebParticipant participant = participants.createParticipant("web.participant" + (i + 1), ops);
 
             try
             {
diff --git a/src/test/java/org/jitsi/meet/test/web/WebTestBase.java b/src/test/java/org/jitsi/meet/test/web/WebTestBase.java
index 2c65f87..24a0e83 100644
--- a/src/test/java/org/jitsi/meet/test/web/WebTestBase.java
+++ b/src/test/java/org/jitsi/meet/test/web/WebTestBase.java
@@ -362,30 +362,6 @@ public class WebTestBase
         return joinParticipantAndWait(2, null, null);
     }
 
-    /**
-     * Joins the "next" participant. Creating a participant and joining a
-     * conference takes significant time. This method can be run in parallel
-     * in order to create multiple participants as quickly as possible. The new
-     * participant will be inserted last in the participant list.
-     *
-     * Note that setting a configuration prefix for the created participant is
-     * currently not supported. If that becomes necessary in the future, a
-     * parameter can be added to this method call hierarchy.
-     *
-     * @param jitsiMeetUrlmeet a {@link JitsiMeetUrl} which represents the full
-     * conference URL which includes server, conference parameters and
-     * the config part.
-     * @param participantOptions the options to be used when creating the
-     * participant.
-     * @return the participant which was created.
-     */
-    public WebParticipant joinNextParticipant(
-        JitsiMeetUrl jitsiMeetUrlmeet,
-        ParticipantOptions participantOptions)
-    {
-        return joinParticipantAndWait(-1, jitsiMeetUrlmeet, participantOptions);
-    }
-
     /**
      * Closes the info dialog for all participants if it is displayed.
      */
```

**Pull request merged**